clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -941,21 +881,108 @@ class KernelObjVisitor {
   // FIXME: Can this be refactored/handled some other way?
   template <typename ParentTy, typename... Handlers>
   void VisitStreamRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
-                         CXXRecordDecl *Wrapper, Handlers &... handlers) {
+                         CXXRecordDecl *Wrapper, QualType RecordTy,
+                         Handlers &... handlers) {
     (void)std::initializer_list<int>{
-        (handlers.enterStruct(Owner, Parent), 0)...};
+        (handlers.enterStream(Owner, Parent, RecordTy), 0)...};
     for (const auto &Field : Wrapper->fields()) {
       QualType FieldTy = Field->getType();
-      (void)std::initializer_list<int>{
-          (handlers.enterField(Wrapper, Field), 0)...};
       // Required to initialize accessors inside streams.
       if (Util::isSyclAccessorType(FieldTy))
         KF_FOR_EACH(handleSyclAccessorType, Field, FieldTy);
-      (void)std::initializer_list<int>{
-          (handlers.leaveField(Wrapper, Field), 0)...};
     }
     (void)std::initializer_list<int>{
-        (handlers.leaveStruct(Owner, Parent), 0)...};
+        (handlers.leaveStream(Owner, Parent, RecordTy), 0)...};
+  }
+
+  template <typename... Handlers>
+  void VisitArrayElementImpl(const CXXRecordDecl *Owner, FieldDecl *ArrayField,
```

Comment:
I feel we have some problems with code style in lots of places

Reply:
This is after clang-format, can you clarify?

Reply:
I meant problems with names, i.e. function names should start with lower case letter, variable names start with upper case letter and so on. clang-format doesn't care about it.

Reply:
Ah, I see.  I'll make a run through this patch myself, then have you correct anything I missed.